### PR TITLE
feat: logging package

### PR DESCRIPTION
**Pull request by winprn**
This package contains a implementation of `BrowserTransport` for browser usage.
Ref: [BrowserConsole](https://github.com/MarcoMedrano/winston-transport-browserconsole/blob/master/src/lib/BrowserConsole.ts)

### CODE REVIEW

File: packages/logger/asconfig.json
Comment (d-roak) [inline packages/logger/asconfig.json:1]: can drop this file

File: packages/logger/package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  ],
+  "exports": {
+    ".": {
+      "types": "./dist/src/index.d.ts",
+      "import": "./dist/src/index.js"
+    },
+    "./wasm": {
+      "types": "./dist/src/index.d.ts",
+      "import": "./src/index.asc.ts"
+    }
```
Comment (d-roak) [inline packages/logger/package.json:25]: can remove it

File: packages/logger/package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  },
+  "scripts": {
+    "asbuild": "yarn asbuild:debug && yarn asbuild:release",
+    "asbuild:debug": "asc --config asconfig.json --target debug",
+    "asbuild:release": "asc --config asconfig.json --target release",
```
Comment (d-roak) [inline packages/logger/package.json:30]: can remove it

File: packages/logger/package.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    "build": "tsc -b",
+    "clean": "rm -rf dist/ node_modules/",
+    "prepack": "tsc -b",
+    "test": "vitest"
+  },
+  "devDependencies": {
+    "assemblyscript": "^0.27.29"
+  },
```
Comment (d-roak) [inline packages/logger/package.json:38]: can remove it

File: packages/logger/src/browser.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+			this.emit("logged", entry);
+		});
+
+		const { message, level } = entry;
+		const mappedMethod = this.methods[
+			level as keyof typeof this.methods
+		] as keyof Console;
+
+		if (Object.getOwnPropertySymbols(entry).length >= 2) {
+			// @ts-ignore
```
Comment (d-roak) [inline packages/logger/src/browser.ts:31]: why ignore the ts compilation stuff?

File: packages/logger/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+	"compilerOptions": {
+		"outDir": "dist"
+	},
+	"references": [
+		{
+			"path": "../object"
+		}
+	],
+	"include": ["src/**/*.ts"],
+	"exclude": ["src/**/*.asc.ts"]
```
Comment (d-roak) [inline packages/logger/tsconfig.json:12]: can rm it

File: packages/logger/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{
+	"extends": "../../tsconfig.json",
+	"compilerOptions": {
+		"outDir": "dist"
+	},
+	"references": [
+		{
+			"path": "../object"
+		}
+	],
```
Comment (d-roak) [inline packages/logger/tsconfig.json:10]: can remove it

File: packages/logger/src/browser.ts (hunk 4 above)
Comment (winprn) [reply packages/logger/src/browser.ts:31]: I tried some workaround but it seems like this is a problem with TypeScript: https://github.com/Microsoft/TypeScript/issues/24587. The last comment said that the issue was resolved in `4.4.0-dev.20210627`, but somehow it still persists.
If you want to I can do some type conversion like: `entry[Object.getOwnPropertySymbols(entry)[1] as unknown as string];` to remove the ignore.

For the other three, I want to access the key using `[]` notation but it wont let me do that

Comment (d-roak) [reply packages/logger/src/browser.ts:31]: `as keyof winston.LogEntry` should do the trick (or something similar)

**Comment by winprn** [comment]
For the current logging in `package/network`, here're some screenshots about the change:
### Browser
image

### CLI
image

File: packages/network/src/node.ts [hunk 7]
```diff
@@ -52,6 +56,7 @@ export class TopologyNetworkNode {
 
 	constructor(config?: TopologyNetworkNodeConfig) {
 		this._config = config;
+		log = new Logger("topology::network", config?.log_config?.level);
```
Comment (d-roak) [inline packages/network/src/node.ts:59]: pass the `log_config` object instead of just the level

File: packages/node/src/rpc/index.ts [hunk 8]
```diff
@@ -13,6 +14,11 @@ import type {
 } from "../proto/rpc_pb.js";
 
 export function init(node: TopologyNode) {
+	const log = new Logger(
+		"topology::rpc",
+		node.config?.network_config?.log_config,
```
Comment (d-roak) [inline packages/node/src/rpc/index.ts:19]: here the same

**Review by d-roak — APPROVED** [review]
lgtm

**Review by d-roak — APPROVED** [review]
lgtm

**Pull request merged**